### PR TITLE
docs: add more details to global styles docs

### LINKED ISSUE #7499: [docs] Further clarify how global styles work
Global styles work in different ways depending on the use case, so we want to capture this in more detail.

1. If every page is wrapped with the same layout component, importing a CSS file or including CSS-in-JS global styles works just fine
2. If every page is _not_ wrapped with the same layout component, you can:
    - with regular CSS, include the stylesheet in `gatsby-browser.js`
    - with CSS-in-JS, every page component needs to include the global styles (or a component that does so)

### PR DESCRIPTION
**Pull request by jlengstorf**
closes #7499

### CODE REVIEW

File: docs/docs/creating-global-styles.md [hunk 1]
```diff
@@ -2,7 +2,124 @@
 title: Creating global styles
 ---
 
-In this guide, we'll walk through using traditional global styles in your site.
+In nearly every site, there will be some global styles, such as a reset or typography defaults. This guide will walk through how to add global styles to your site, whether you use standard `.css` files or a CSS-in-JS solution.
```
Comment (KyleAMathews) [inline docs/docs/creating-global-styles.md]: standard css files or w/ preprocessor. Probably should clarify this as preprocessors are so common.

File: docs/docs/creating-global-styles.md [hunk 2]
```diff
@@ -2,7 +2,124 @@
...
+
+## How to add global styles in Gatsby with standard CSS files
+
+In general, the best way to add global styles is with a [shared layout component](https://next.gatsbyjs.org/tutorial/part-three/#your-first-layout-component). This layout will be used in every page component to apply global styles (and often shared components such as a header and/or footer).
```
Comment (KyleAMathews) [inline docs/docs/creating-global-styles.md]: Doesn't need "In general".

Comment (KyleAMathews) [reply docs/docs/creating-global-styles.md]: *This layout component

Comment (KyleAMathews) [reply docs/docs/creating-global-styles.md]: The second sentence is a bit awkward. I'd frame the layout component as a component for things that's shared across different pages so stuff like styles & headers, etc.

File: docs/docs/creating-global-styles.md [hunk 3]
```diff
@@ -2,7 +2,130 @@
...
+
+- [How to add global styles in Gatsby with standard CSS files](#how-to-add-global-styles-in-gatsby-with-standard-css-files)
+- [How to add global styles in Gatsby using CSS-in-JS](#how-to-add-global-styles-in-gatsby-using-css-in-js)
+- [Add global styles with CSS files and no layout component](#add-global-styles-with-css-files-and-no-layout-component)
+
+## How to add global styles in Gatsby with standard CSS files
+
+The best way to add global styles is with a [shared layout component](https://next.gatsbyjs.org/tutorial/part-three/#your-first-layout-component). This layout component is used for things that are shared throughout the site, including styles, header components, and other common items.
+
+> _NOTE:_ This pattern is implemented by default in [the default starter](https://github.com/gatsbyjs/gatsby-starter-default).
```
Comment (amberleyromo) [inline docs/docs/creating-global-styles.md]: Not a big deal, but could link directly to the layout file here instead.

File: docs/docs/creating-global-styles.md [hunk 4]
```diff
@@ -2,7 +2,130 @@
...
+export default () => <Layout>Hello world!</Layout>
+```
+
+Run `gatsby develop` and you’ll see the global styles applied.
+
+![Global styles](./images/global-styles.png)
+
+## How to add global styles in Gatsby using CSS-in-JS
+
+> **NOTE:** For this example, we’ll be using [Emotion](https://emotion.sh), but the implementation is similar for other CSS-in-JS solutions as well.
```
Comment (amberleyromo) [inline docs/docs/creating-global-styles.md]: The first "note" is italicized, this is bold. We don't have consistency across docs rn, but maybe pick one within this file?

File: docs/docs/creating-global-styles.md (hunk 3 above)
Comment (jlengstorf) [reply docs/docs/creating-global-styles.md]: Good call!

**Comment by shannonbux** [comment]
Huh, where is this doc in the navigation, @jlengstorf? I can't find it!

**Comment by fk** [comment]
@shannonbux https://next.gatsbyjs.org/docs/creating-global-styles/

**Pull request merged**